src/compile_cache.cc

```diff
@@ -223,13 +226,93 @@ void CompileCacheHandler::ReadCacheFile(CompileCacheEntry* entry) {
   Debug(" success, size=%d\n", total_read);
 }
 
+#ifdef _WIN32
+constexpr bool IsWindowsDeviceRoot(const char c) noexcept {
+  return (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z');
+}
+#endif
+
+static std::string NormalisePath(std::string_view path) {
+  std::string normalised_string(path);
+  constexpr std::string_view file_scheme = "file://";
+  if (normalised_string.rfind(file_scheme, 0) == 0) {
+    normalised_string.erase(0, file_scheme.size());
+  }
```

Comment:
Hmm, if this is a file URL then just stripping the protocol is going to miss edge cases. I think what you'll actually want is to use ada-url to parse the url then get the `pathname` component after parsing (otherwise you risk grabbing hostname, query string, fragment identifiers, etc. It would also be necessary to decode any percent-encoded characters in the pathname.

Reply:
then we should drop the `file://` handling entirely if it is not going to be handled correctly here.

Reply:
Sorry, I'm going to have to insist. Even if it's just an edge case, intentionally using a buggy implementation that we know will have issues is not what we should be doing -- we already have enough bugs with path handling in various apis we don't need to intentionally add more. If we're going to handle `file://` urls here at all we should do so correctly.

Reply:
Thank you @Aditi-1400 :-) I appreciate that. I should have an opportunity to review in the next day or two.